app/behaviour.py

```
@@ -101,9 +101,15 @@ def __test_strategies(self, market_data):
 
                         else:
                             self.logger.warn("No such behaviour %s, skipping.", behaviour)
+                except ValueError as e:
```

I think I mentioned this before, but we should probably let the ValueError be thrown here.

---

You did, but I remembered why we actually need that.
1. We are raising a ValueError when looking up the time_unit of a pair, if we don't handle it then the application breaks instead of the pair being skipped.

2. Some exchanges return junk OHLCV sticks with None types which again raises an error and breaks the app instead of skipping to the next pair.

3. We are raising a ValueError when the historical_data returned by the exchange is completely empty as this breaks the pandas df conversion. Which should also skip the pair not break the app.

---

Okay, I agree that 2 & 3 should not stop the application. But if the user enters an invalid timeframe, keeping the app running doesn't really serve a purpose as it won't be able to return any data. I still believe that #1 should kill the program

---

Ideally it would skip the exchange, rather than kill the program... because if they have enabled multiple exchanges it might be valid for one of them.